server/reload.go

```diff
@@ -313,8 +323,17 @@ func (c *clusterOption) Apply(s *Server) {
 	}
 	s.setRouteInfoHostPortAndIP()
 	s.mu.Unlock()
-	if c.newValue.Name != "" && c.newValue.Name != s.ClusterName() {
-		s.setClusterName(c.newValue.Name)
+
+	// Check whether the cluster name has been removed or changed
+	// as that could affect a cluster membership.
+	switch {
+	case c.oldValue.Name != "" && c.newValue.Name == "":
+		s.setClusterName(s.defaultClusterName, true)
```

Comment:
No action required, just a comment: this is immutable I think, so should not report data races. Otherwise, you could always capture the default cluster name while under the server lock few lines above.